code/Model/SiteTree.php

```diff
@@ -552,12 +552,12 @@ public static function page_type_classes()
     {
         $classes = ClassInfo::getValidSubClasses();
 
-        $baseClassIndex = array_search(self::class, $classes ?? []);
+        $baseClassIndex = array_search(SiteTree::class, $classes ?? []);
         if ($baseClassIndex !== false) {
             unset($classes[$baseClassIndex]);
         }
 
-        $kill_ancestors = self::config()->get('hide_pagetypes', Config::UNINHERITED) ?? [];
+        $kill_ancestors = static::config()->get('hide_pagetypes', Config::UNINHERITED) ?? [];
```

Comment:
```suggestion
        $kill_ancestors = SiteConfig::config()->get('hide_pagetypes', Config::UNINHERITED) ?? [];
```
This is a [new config as of 5.2](https://github.com/silverstripe/silverstripe-cms/pull/2902) and was very much intended to be just a single config value across the whole class hierarchy.

Reply:
Updated